src/core/continuous_space.jl

```diff
@@ -352,16 +352,23 @@ The argument `method` provides three pairing scenarios
   agent is paired to its nearest neighbor. Similar to `:nearest`, each agent can belong
   to only one pair. This functionality is useful e.g. when you want some agents to be
   paired "guaranteed", even if some other agents might be nearest to each other.
+- `:types`: For mixed models only. Return every pair of agents within radius `r` (similar
+  to `:all`), with an additional constraint that excludes neighbors with the same type.
+  For example, a mixed model of `Union{Sheep,Wolf,Grass}`, pairs of `(Sheep,Sheep)` will
+  be excluded. To also exclude `(Wolf,Grass)` pairs, provide a scheduler that samples only

```

Comment:
I don't immediatelly understand how this exclusion would work.

Reply:
Take a look at the tests, I have a model with `Union{Agent6, AgentU1, AgentU2}`, which wont match `(Agent6, Agent6)` or pairs of the other types. Then provide a scheduler that returns allowed matches, such that `(Agent6, AgentU1)` would not be possible.

The predator model, you would exclude `Grass` in the scheduler to only allow `(Sheep,Wolf)`.

Generally speaking you may have `Union{Road, House, River, Boat, Car, Bicycle}`, when you need only interactions on the road. Your scheduler returns ids of `Road, Car, Bicycle`, with interactions of `(Road, Car), (Road, Bicycle), (Car, Bicycle)` possible.

Reply:
I do not understand why interaction `(Car, Bicycle)` is not possible though....?

Reply:
It is, that would be the point of this. If you only wanted `(Road, Car), (Road, Bicycle)`, that would require two calls.